File: server/src/main/java/com/hospitaltender/server/service/TenderItemService.java
package com.hospitaltender.server.service;

import com.hospitaltender.server.dto.request.CreateTenderItemRequest;
import com.hospitaltender.server.dto.request.UpdateTenderItemRequest;
import com.hospitaltender.server.dto.response.TenderItemResponse;

import java.util.List;

public interface TenderItemService {

    TenderItemResponse create(CreateTenderItemRequest request);

    TenderItemResponse getById(Long id);

    List<TenderItemResponse> getByTenderId(Long tenderId);

    TenderItemResponse update(Long id, UpdateTenderItemRequest request);

    void delete(Long id);
}

File: ui/hospital-system-ui/src/services/tender-item.service.ts
import api from './api'
import { TenderItemResponse, CreateTenderItemRequest, UpdateTenderItemRequest } from '@/types'

export const tenderItemService = {
  getByTenderId: (tenderId: number) =>
    api.get<TenderItemResponse[]>(`/tender-items/tender/${tenderId}`).then((res) => res.data),

  getById: (id: number) =>
    api.get<TenderItemResponse>(`/tender-items/${id}`).then((res) => res.data),

  create: (data: CreateTenderItemRequest) =>
    api.post<TenderItemResponse>('/tender-items', data).then((res) => res.data),

  update: (id: number, data: UpdateTenderItemRequest) =>
    api.put<TenderItemResponse>(`/tender-items/${id}`, data).then((res) => res.data),

  delete: (id: number) =>
    api.delete(`/tender-items/${id}`),
}
